### PR TITLE
feat(DevServer): batch bundles & run them asynchronously

### LINKED ISSUE #14913: bake: assertion failure with invalid cache key
### What version of Bun is running?

1.1.34 (debug build of latest main branch)

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

This is an issue that only occurs when running a debug, and under certain circumstances.

Reproduction

1. create a new minimal bun bake project
2. add a folder; `pages`
3. create a file; `pages/route.tsx`, with a proper default exported component
4. add above file in bun bake's index file `routes` (as below)
5. run the application `bun-debug run index.tsx`
6. add a new folder in the `pages` directory

### What is the expected behavior?

nothing happens, app continues to run as before

### What do you see instead?

app crashes with error:

```sh
panic: Internal Assertion Failure: Invalid cache key "/bun-bake/pages/" See Resolver.assertValidCacheKey for details.
```

### Additional information

sample config:

```tsx
wipDevServerExpectHugeBreakingChanges({
  // routes must be statically known so production
  // builds can prepare all required assets.
  routes: [
    { pattern: '/', entrypoint: './pages/route.tsx' },
  ],
  // for framework authors, attach your framework
  // via the interface described in `bake.d.ts`
  framework: 'react',
});
```

ping @paperdave

### PR DESCRIPTION
**Pull request by paperclover**
### What does this PR do?

Dev:
- Remove `waitFor` calls, fixing a crash. there is now a "current bundle" and a bundle queue, which hold references to the requests pending on them. once the current bundle is finished, the "next bundle" is drained. the cool part of this is every bundle attempted is combined, deduplicated, and run all at once.
- Document the atomics hell
- Closes #14913
- Enable console buffering after printing reload message
- Print a log when routes successfully bundle
- Timer for reloads starts much earlier (timer is started on the watch thread); have not tested release build but if we continue showing 1ms then that is much more impressive than 1ms on previous branch.
- Fix most but not all of the CSS reloading issues
- Some usability bugs while testing

Prod:
- Using the same entry point in a client and server part will bundle properly.

All:
- Fixes parsing "style" from framework api

React integration:
- Fix flash of unstyled content
- Support view transitions
- Back button uses an in memory cache of the page to avoid re-fetching

Non-bake
- Bun.build supports plugins with `loader: 'object'`

### CODE REVIEW

**Comment by robobun** [comment]
@paperdave, your commit 4e5efc388ae841aab6b332f74b981bc8a92208e9 is building: [`#6249`](https://buildkite.com/bun/bun/builds/6249)

File: src/bun.js/api/JSBundler.zig [hunk 1]
```diff
@@ -1041,9 +1045,13 @@ pub const JSBundler = struct {
             JSC.markBinding(@src());
```
Comment (dylan-conway) [inline src/bun.js/api/JSBundler.zig:1045]: `JSError!JSValue` return type

File: src/bundler/bundle_v2.zig [hunk 2]
```diff
@@ -2072,19 +2059,25 @@ pub const BundleV2 = struct {
...
         this.unique_key = generateUniqueKey();
 
         this.graph.heap.helpCatchMemoryIssues();
 
-        var css_entry_points: std.AutoArrayHashMapUnmanaged(Index, BakeBundleOutput.CssEntryPointMeta) = .{};
+        var ctx: BakeBundleStart = .{ .css_entry_points = .{} };
         this.graph.pool.pool.schedule(try this.enqueueEntryPoints(.dev_server, .{
             .files = bake_entry_points,
-            .css_data = &css_entry_points,
+            .css_data = &ctx.css_entry_points,
```
Comment (dylan-conway) [inline src/bundler/bundle_v2.zig:2071]: is `css_data` going to be used outside the scope of this function?

File: src/js_parser.zig [hunk 3]
```diff
@@ -5934,10 +5932,9 @@ fn NewParser_(
                         }
                     }
 
-                    return p.newExpr(E.Dot{
-                        .target = p.newExpr(E.Identifier.init(ns_alias.namespace_ref), loc),
-                        .name = ns_alias.alias,
-                        .name_loc = loc,
+                    return p.newExpr(E.ImportIdentifier{
```
Comment (dylan-conway) [inline src/js_parser.zig:5935]: What is this fixing?

**Review by dylan-conway — APPROVED** [review]

**Pull request merged**